OptimizelySDK/Bucketing/DecisionService.cs

```diff
@@ -70,61 +70,63 @@ public DecisionService(Bucketer bucketer, IErrorHandler errorHandler, UserProfil
-        #else
+#else
             ForcedVariationMap = new System.Collections.Concurrent.ConcurrentDictionary<string, Dictionary<string, string>>();
-        #endif
+#endif
         }
 
         /// <summary>
         /// Get a Variation of an Experiment for a user to be allocated into.
         /// </summary>
         /// <param name = "experiment" > The Experiment the user will be bucketed into.</param>
-        /// <param name = "userId" > The userId of the user.
-        /// <param name = "filteredAttributes" > The user's attributes. This should be filtered to just attributes in the Datafile.</param>
+        /// <param name = "user" > Optimizely user context.
+        /// <param name = "config" > Project config.</param>
         /// <returns>The Variation the user is allocated into.</returns>
         public virtual Result<Variation> GetVariation(Experiment experiment,
-            string userId,
-            ProjectConfig config,
-            UserAttributes filteredAttributes)
+            OptimizelyUserContext user,
+            ProjectConfig config)
         {
-            return GetVariation(experiment, userId, config, filteredAttributes, new OptimizelyDecideOption[] { });
+            return GetVariation(experiment, user, config, new OptimizelyDecideOption[] { });
         }
 
         /// <summary>
         /// Get a Variation of an Experiment for a user to be allocated into.
         /// </summary>
         /// <param name = "experiment" > The Experiment the user will be bucketed into.</param>
-        /// <param name = "userId" > The userId of the user.
-        /// <param name = "filteredAttributes" > The user's attributes. This should be filtered to just attributes in the Datafile.</param>
+        /// <param name = "user" > optimizely user context.
+        /// <param name = "config" > Project Config.</param>
+        /// <param name = "options" >An array of decision options.</param>
         /// <returns>The Variation the user is allocated into.</returns>
         public virtual Result<Variation> GetVariation(Experiment experiment,
-            string userId,
+            OptimizelyUserContext user,
             ProjectConfig config,
-            UserAttributes filteredAttributes,
             OptimizelyDecideOption[] options)
         {
             var reasons = new DecisionReasons();
+            var userId = user.GetUserId();
             if (!ExperimentUtils.IsExperimentActive(experiment, Logger))
                 return Result<Variation>.NullResult(reasons);
 
             // check if a forced variation is set
-            var forcedVariationResult = GetForcedVariation(experiment.Key, userId, config);
-            reasons += forcedVariationResult.DecisionReasons;
+            var decisionVariation = GetForcedVariation(experiment.Key, userId, config);
+            reasons += decisionVariation.DecisionReasons;
+            var variation = decisionVariation?.ResultObject;
```

Comment:
null checking("?") not consistent 114-115. I see these in several places. Please review carefully.